src/Workspaces/CSharp/Portable/Utilities/CSharpUseImplicitTypeHelper.cs

```diff
@@ -259,9 +259,31 @@ protected override bool ShouldAnalyzeForEachStatement(ForEachStatementSyntax for
             }
 
             // variables declared using var cannot be used further in the same initialization expression.
-            if (initializer.DescendantNodesAndSelf()
-                    .Where(n => (n as IdentifierNameSyntax)?.Identifier.ValueText.Equals(identifier.ValueText) == true)
-                    .Any(n => semanticModel.GetSymbolInfo(n, cancellationToken).Symbol?.IsKind(SymbolKind.Local) == true))
+            bool isReusedInInitializer(SyntaxNode n)
```

Comment:
Please don't plae nested function in-situ with the rest of the code.  It makes it very hard to read and grok the outer code flow.  If you want to use a nested function, please place after hte last 'return' in the method.

Reply:
In combination with your proposal in https://github.com/dotnet/roslyn/pull/27897#discussion_r195865680, let's inline it for now. 👍